openmc/stats/multivariate.py

```diff
@@ -613,6 +616,123 @@ def from_xml_element(cls, elem):
+    ----------
+    elem_weight_scheme : String, optional
+        Weighting scheme for sampling element from mesh, defaults to volume
+    mesh_id : int, optional
+        The ID of the mesh, defaults to 1
+    weights_from_file : Iterable of Real, optional
+        A list of values which represent the weights of each element
+
+    """
+
+    def __init__(self, mesh, elem_weight_scheme="volume", weights_from_file=[0.0]):
+        self.mesh = mesh        
+        self.elem_weight_scheme = elem_weight_scheme
+        self.mesh_id = self.mesh.id
+        self.weights_from_file = weights_from_file
+
+    @property
+    def mesh(self):
+        return self._mesh
+
+    @mesh.setter
+    def mesh(self, mesh):
+        if mesh != None:
+            cv.check_type('Unstructured Mesh', mesh, MeshBase)
+        self._mesh = mesh
+
+    @property
+    def elem_weight_scheme(self):
+        return self._elem_weight_scheme
+
+    @elem_weight_scheme.setter
+    def elem_weight_scheme(self, elem_weight_scheme):
+        cv.check_value('Scheme for sampling an element from the mesh', elem_weight_scheme, ['volume', 'file'])
+        self._elem_weight_scheme = elem_weight_scheme
+
+    @property
+    def mesh_id(self):
+        return self._mesh_id
+
+    @mesh_id.setter
+    def mesh_id(self, mesh_id):
+        cv.check_type('Mesh file ID number', mesh_id, int)
+        self._mesh_id = mesh_id
+
+    @property
+    def weights_from_file(self):
+        return self._weights_from_file
+
+    @weights_from_file.setter
+    def weights_from_file(self, given_weights):
+        self._weights_from_file = (np.array(given_weights, dtype=float)).flatten()
+
+    @property
+    def num_weight_bins(self):
+        if self.weights_from_file == None:
+            raise ValueError('Weight bins are not set')
+        return self.weights_from_file.size
+                
+    def to_xml_element(self):
+        """Return XML representation of the spatial distribution
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing spatial distribution data
+
+        """
+        element = ET.Element('space')
+        element.set('type', 'mesh')
+        element.set("elem_weight_scheme", self.elem_weight_scheme)
+        element.set("mesh_id", str(self.mesh_id))
+
+        subelement = ET.SubElement(element, 'weights_from_file')
+        subelement.text = ' '.join(str(e) for e in self.weights_from_file)
+
+        return element
+
+    @classmethod
+    def from_xml_element(cls, elem):
+        """Generate spatial distribution from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.stats.MeshIndependent
+            Spatial distribution generated from XML element
+
+        """
+
+        elem_weight_scheme = elem.get('elem_weight_scheme')
+        mesh_id = int(elem.get('mesh_id'))
```

Comment:
We'll want to parse the mesh from the settings file as well. This section in `weight_windows.py` shows how you can do that: 

https://github.com/openmc-dev/openmc/blob/a4d6de1d7b5dc45812c1ba3ec4f9aaec86767eda/openmc/weight_windows.py#L330-L335